rq/queue.py

```diff
@@ -310,10 +322,64 @@ def create_job(self, func, args=None, kwargs=None, timeout=None,
 
         return job
 
+    def setup_dependencies_and_enqueue(
```

Comment:
One more small change, can you change this to `setup_dependencies` and leave out the `enqueue_job` line? It's cleaner to have a single method to setup dependencies and we can potentially reuse this in other places.

Reply:
Unless the `enqueue_job` is part of the logic and it's hard to untangle, in which case we can leave it as it is.

Reply:
@selwin I thought about doing that, but that means we could have the case if `depends_on` is True and a job is found we would have that job that may already be in the registry of the queue, or the job got no where at all

Reply:
@selwin Actually, on second thought, there should be a way to refactor it out.  Maybe we should have a discussion on the best way how?   Because if aren't enqueueing we would need to know if the job was deferred or if none were found and should just be enqueued maybe via a bool return value?

Reply:
What if we check the job status? If `job.status` is not `DEFERRED`, then it should be enqueued.

Reply:
@selwin Done.  Should I be increasing the version number to `1.9.0`?:
https://github.com/rq/rq/blob/5620f2932cffb31df58a164d78e5cd4327b89414/rq/version.py#L5

Reply:
Yes, would be great if you could do that. After this is merged in, I'll also pull in https://github.com/rq/rq/pull/1480 and cut a release in a few weeks.

Reply:
@selwin Awesome, thank you so much!  I have updated the version to 1.9.0.